pyvista/plotting/plotting.py

```diff
@@ -1299,8 +1299,11 @@ def add_mesh(self, mesh, color=None, style=None, scalars=None,
                 raise TypeError(f'Object type ({type(mesh)}) not supported for plotting in PyVista.')
         ##### Parse arguments to be used for all meshes #####
 
+        # Avoid mutating input
         if scalar_bar_args is None:
             scalar_bar_args = {'n_colors': n_colors}
+        else:
+            scalar_bar_args = scalar_bar_args.copy()
```

Comment:
Just for consideration: the `scalar_bar_args` can contain mutable values (colour lists, mappers), so technically a deepcopy would be more correct. But we'd only add/change an immutable value, so a shallow copy should be fine. As it's mostly shown by your new test.

Reply:
As far as I can tell those mutable values are untouched in `add_mesh` and `add_volume`.  They _may_ be changed inside `add_scalar_bar` (I haven't looked), but in that case we should handle it there.